android/app/src/main/java/app/organicmaps/car/screens/RequestPermissionsScreen.java

```diff
@@ -0,0 +1,87 @@
+package app.organicmaps.car.screens;
+
+import static android.Manifest.permission.ACCESS_COARSE_LOCATION;
+import static android.Manifest.permission.ACCESS_FINE_LOCATION;
+
+import androidx.annotation.NonNull;
+import androidx.car.app.CarContext;
+import androidx.car.app.model.Action;
+import androidx.car.app.model.CarIcon;
+import androidx.car.app.model.MessageTemplate;
+import androidx.car.app.model.ParkedOnlyOnClickListener;
+import androidx.car.app.model.Template;
+import androidx.core.graphics.drawable.IconCompat;
+import androidx.lifecycle.LifecycleOwner;
+
+import app.organicmaps.R;
+import app.organicmaps.car.screens.base.BaseScreen;
+import app.organicmaps.car.util.Colors;
+import app.organicmaps.car.util.LocationUtils;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class RequestPermissionsScreen extends BaseScreen
+{
+  public interface PermissionsGrantedCallback
+  {
+    void onPermissionsGranted();
+  }
+
+  private static final List<String> LOCATION_PERMISSIONS = Arrays.asList(ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION);
+
+  @NonNull
+  private final PermissionsGrantedCallback mPermissionsGrantedCallback;
+
+  public RequestPermissionsScreen(@NonNull CarContext carContext, @NonNull PermissionsGrantedCallback permissionsGrantedCallback)
+  {
+    super(carContext);
+    mPermissionsGrantedCallback = permissionsGrantedCallback;
```

Comment:
Can Screen itself return a result, like Activity? Even if not – I suggest checking location permissions in the caller when this dialog is finished instead of using callbacks.

Reply:
https://github.com/android/car-samples/blob/main/car_app_library/navigation/common/src/main/java/androidx/car/app/sample/navigation/common/car/RequestPermissionScreen.java

Reply:
https://developer.android.com/reference/androidx/car/app/Screen#setResult(java.lang.Object) does this method work?

Reply:
See https://developer.android.com/reference/androidx/car/app/ScreenManager#pushForResult(androidx.car.app.Screen,androidx.car.app.OnScreenResultListener)

Auto frameworks provides this approach for showing screens for results. Why should we follow that callback-based approach from the example?